crates/sui/tests/cli_tests.rs

```diff
@@ -1780,6 +1851,167 @@ async fn test_package_upgrade_command() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[sim_test]
+async fn test_package_management_on_upgrade_command() -> Result<(), anyhow::Error> {
```

Comment:
Largely a copy of `test_package_upgrade_command` but with important changes to check the right version / addresses are returned from the lock contents